store/commitment/store.go

```diff
@@ -0,0 +1,151 @@
+package commitment
+
+import (
+	"errors"
+	"fmt"
+
+	dbm "github.com/cosmos/cosmos-db"
+	ics23 "github.com/cosmos/ics23/go"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/v2"
+	"cosmossdk.io/store/v2/commitment/iavl"
+	"cosmossdk.io/store/v2/commitment/types"
+)
+
+const (
+	// DefaultTreeType is the default tree type.
+	DefaultTreeType = "default"
+	IavlTreeType    = "iavl"
+)
+
+var _ store.Committer = (*CommitStore)(nil)
+
+// CommitStore is a wrapper around the state Tree.
+type CommitStore struct {
+	multiTrees map[string]types.Tree
+}
+
+// NewCommitStore creates a new CommitStore instance.
+func NewCommitStore(storeConfigs map[string]interface{}, db dbm.DB, logger log.Logger) (*CommitStore, error) {
```

Comment:
nit: use `any` over `interface{}` -- its shorter and cleaner to read :)

```suggestion
func NewCommitStore(storeConfigs map[string]any, db dbm.DB, logger log.Logger) (*CommitStore, error) {
```